app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,84 @@
+class ArticlesController < ApplicationController
+  before_action :set_article, only: %i[ show edit update destroy count_comments set_article_number_of_comments]
+
+  # GET /articles or /articles.json
+  def index
+    @articles = Article.all
+  end
+
+  # GET /articles/1 or /articles/1.json
+  def show
+  end
+
+  # GET /articles/new
+  def new
+    @article = Article.new
+  end
+
+  # GET /articles/1/edit
+  def edit
+  end
+
+  # POST /articles or /articles.json
+  def create
+    @article = Article.new(article_params)
+    @article.title = CommentsCountAnalyzer.get_article_name(@article.url_of_article)
+
+    respond_to do |format|
+      if @article.save
+        format.html { redirect_to article_url(@article), notice: "Article was successfully created." }
+        format.json { render :show, status: :created, location: @article }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @article.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /articles/1 or /articles/1.json
+  def update
+    respond_to do |format|
+      if @article.update(article_params)
+        format.html { redirect_to article_url(@article), notice: "Article was successfully updated." }
+        format.json { render :show, status: :ok, location: @article }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @article.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /articles/1 or /articles/1.json
+  def destroy
+    @article.destroy
+
+    respond_to do |format|
+      format.html { redirect_to articles_url, notice: "Article was successfully destroyed." }
+      format.json { head :no_content }
+    end
+  end
+
+    # GET /articles/1 or /articles/1.json
+  def count_comments
+    comments_count = CommentsCountAnalyzer.perform(@article)
+
+    respond_to do |format|
+      format.html { redirect_to article_url(@article) }
+      format.json { render :show, status: :ok, location: @article }
+    end
+
+    @article.update_attribute(:number_of_comments, comments_count)
```

Comment:
just use `update` method and better do it before `respond_to`. also send ok response if `update` returns true